### PR TITLE
CIRC-575 BE: In transit report: check in fields

### PR DESCRIPTION
**Pull request by Svitlana-Zmiivska**
## Purpose
Add check in fields to a transit report.
Resolves [CIRC-575](https://issues.folio.org/browse/CIRC-575)

## Approach
- get check in fields from an  item and add them to a report;

## Links
[CIRC-575](https://issues.folio.org/browse/CIRC-575)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/representations/ItemReportRepresentation.java [hunk 1]
```diff
@@ -60,9 +61,24 @@ public JsonObject createItemReport(InTransitReportEntry inTransitReportEntry) {
     if (loan != null) {
       writeLoan(itemReport, loan);
     }
+
+    final LastCheckIn lastCheckIn = item.getLastCheckIn();
+    if (lastCheckIn != null) {
+      writeLastCheckIn(itemReport, lastCheckIn);
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/representations/ItemReportRepresentation.java]: This suggests that new properties have been added to the item representation in the report. If so, please can the interface version be updated (in the descriptor and RAML file) and the relevant schema updated.

File: src/main/java/org/folio/circulation/resources/ItemsInTransitResource.java [hunk 2]
```diff
@@ -115,8 +115,12 @@ private void getMany(RoutingContext routingContext) {
     return CompletableFuture.completedFuture(Result.succeeded(item))
       .thenComposeAsync(itemRepository::fetchItemRelatedRecords)
       .thenComposeAsync(result -> result
-        .combineAfter(it -> servicePointRepository
-          .getServicePointById(it.getInTransitDestinationServicePointId()), Item::updateDestinationServicePoint));
+        .combineAfter(currentItem -> servicePointRepository
+          .getServicePointById(currentItem.getInTransitDestinationServicePointId()), Item::updateDestinationServicePoint))
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/resources/ItemsInTransitResource.java]: I imagine `it` was used as a shorter name to help with line length. Does this extend over the maximum line length?

File: src/main/java/org/folio/circulation/domain/Item.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public static Item from(JsonObject representation) {
       null,
       null,
       null,
-      null,
+      LastCheckIn.fromItemJson(representation),
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/Item.java]: What happens if there isn't any last check in information?

File: src/main/java/org/folio/circulation/domain/representations/ItemReportRepresentation.java (hunk 1 above)
Comment (Svitlana-Zmiivska) [reply src/main/java/org/folio/circulation/domain/representations/ItemReportRepresentation.java]: Updated

File: src/main/java/org/folio/circulation/resources/ItemsInTransitResource.java (hunk 2 above)
Comment (Svitlana-Zmiivska) [reply src/main/java/org/folio/circulation/resources/ItemsInTransitResource.java]: Split into 2 lines. It seems to me `currentItem ` is a more readable name.

```

**Comment by Svitlana-Zmiivska** [comment]
@marc-johnson Could you please provide more details on the real case when the last check in information is not present in an item and the item has in transit status?

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -22,7 +22,7 @@
     },
     {
       "id": "inventory-reports",
-      "version": "0.2",
+      "version": "0.3",
```
Comment (bohdan-suprun) [inline descriptors/ModuleDescriptor-template.json]: I think we have to update the version in RAML as well.

Comment (Svitlana-Zmiivska) [reply descriptors/ModuleDescriptor-template.json]: Updated

**Comment by Svitlana-Zmiivska** [comment]
@marcjohnson-kint, do you have any other comments?

**Pull request merged**